pkg/xray/watches.go

```diff
@@ -162,24 +200,46 @@ func packProjectResources(resources WatchProjectResources) []interface{} {
 		if len(res.RepoType) > 0 {
 			resourceMap["repo_type"] = res.RepoType
 		}
-		resourceMap["filter"] = packFilters(res.Filters)
+
+		resourceMap = packFilters(res.Filters, resourceMap)
+
 		list = append(list, resourceMap)
 	}
 
 	return list
 }
 
-func packFilters(filters []WatchFilter) []interface{} {
-	var l []interface{}
-	for _, f := range filters {
-		m := map[string]interface{}{
-			"type":  f.Type,
-			"value": f.Value,
+func packFilters(filters []WatchFilter, resources map[string]interface{}) (map[string]interface{}) {
+	var stringFilters []map[string]interface{}
+	var antFilters []map[string]interface{}
+
+	for _, filter := range filters {
```

Comment:
So, the way I prefer to use cases, to eliminate the potential for problems it to wrap it in a closure and call the closure immediately

```go
thing := func() string {
   swtich thing {
   case whatever: 
    return foo
}()
```
instead of assigning variables = this entirely eliminates the potential for misaligning variables